### PR TITLE
Simple build

### PR DESCRIPTION
**Pull request by zerothi**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml
Comment (xhochy) [inline recipe/meta.yaml]: It doesn't look like dependencies actually require `numpy>=1.20`. Can you change to only `numpy` and rerender? If that doesn't work, I can have a look at the error, too.

```suggestion
    - numpy
```

Comment (zerothi) [reply recipe/meta.yaml]: But then I would have 2 separate branches building the same feedstock for py3.6? No?
That is what I am trying to avoid.

Comment (zerothi) [reply recipe/meta.yaml]: I agree, I don't have any requirements for numpy (only >=1.13). But the problem is that I build against numpy API and in numpy 1.20 the internal array has changed (see [here](https://github.com/numpy/numpy/releases/tag/v1.20.0) under C API changes.
I have had users complaining that my latest feedstock (build against 1.20) doesn't work on their setup since they have another package required pre 1.20 numpy versions... :( sigh... ;)

Comment (xhochy) [reply recipe/meta.yaml]: Yes, this will build for the minimal supported `numpy` version on conda-forge (For Python 3.6 this is 1.17 for now) and it should be compatible with all newer NumPy versions (The C-API of NumPy guarantees forward compatibility). 

> But then I would have 2 separate branches building the same feedstock for py3.6?

No, you shouldn't. There should be only a single CI entry.

Comment (zerothi) [reply recipe/meta.yaml]: But my cythonization does not insert the requested operation required to be backward compatible...
So numpy's release page says that 

> Starting with NumPy 1.20 code that is compiled against an earlier
version will be API incompatible with NumPy 1.20. The fix is to either
compile against 1.16.6 (if the NumPy 1.16 release is the oldest release
you wish to support), or manually inline the macro by replacing it with
the new definition:

and I don't have control over this now?

My problem is that users on py3.7 may have a package that requires numpy <1.20 (I don't know why ;) ) and then it will install, but when the users runs my installation, it crashes...

I am not sure exactly how you would suggest me to go forward? (but thanks a lot for your effort!!!)
should I ditch the 2 branch method?

Comment (xhochy) [reply recipe/meta.yaml]: That is a problem I personally reported and is fixed here in conda-forge by simply using the global pin. It is important that you don't compile with `numpy<1.16.6`. I'm not sure what you mean with the two-branch-method but please just keep using the default way.

Comment (zerothi) [reply recipe/meta.yaml]: Thanks! Ok, so I should do something like this:
```yaml

host:
  ...
  - numpy >=1.16.6
run:
  - numpy
```
and that's it?

Comment (xhochy) [reply recipe/meta.yaml]: No, please just apply my suggestion as above and rerender.

Comment (zerothi) [reply recipe/meta.yaml]: as for the two branch model, I was referring to [this](https://conda-forge.org/docs/maintainer/updating_pkgs.html?highlight=branch#maintaining-several-versions) which I successfully use for other feedstocks where I have multiple versions I maintain at the same time. :)

Comment (zerothi) [reply recipe/meta.yaml]: Now I am a bit confused, since then I would have this:
```yaml
host:
 - numpy
run:
  - {{ pin_compatible('numpy', min_pin='x.x') }}
```
but is that what I want? I don't want 1.16.6 to be the minimum version...

Comment (zerothi) [reply recipe/meta.yaml]: I will try only changing the above you mentioned and see.

Comment (xhochy) [reply recipe/meta.yaml]: > as for the two branch model, I was referring to this which I successfully use for other feedstocks where I have multiple versions I maintain at the same time. :)

That is meant for supporting multiple versions of the package, i.e. `sisl` in this case, not different dependency sets. For that you can use `conda_build_config.yaml`. That is though not needed here.

Comment (xhochy) [reply recipe/meta.yaml]:
```
  - {{ pin_compatible('numpy', min_pin='x.x') }}
```

is fine as it will add a `>=1.17,<2` pin automatically to run. You could though omit the `min_pin='x.x'` part as this is not necessary and that default would be OK.

**Comment by xhochy** [comment]
🚀 Python 3.6 passed.

**Comment by zerothi** [comment]
>  Python 3.6 passed.

Great! Thanks a lot. I must say that I don't fully understand how this global pinning works since I don't reflect my actual dependencies in the `meta.yaml`. For instance I do require numpy >=1.13. But I don't mention this anywhere but in the `requirements` file of the project (not anywhere in conda at least)... So the interplay between conda and the python package it-self is not fully clear to me.

I'll try it out once merged, and if you don't hear anything, it works! Thanks a lot again!

**Comment by xhochy** [comment]
You can add your requirement constraints to the `meta.yaml` but should also add plain `- numpy` entries for things that have a global pinning. Then `conda smithy` will auto-generate a local pinning file for the Python+OS combination in the feedstock in the `.ci_support` directory. Especially for `numpy`, the pinning is a bit tricky to get it working on all OSes and for a wide range of users.

For the full global pinning, have a look at https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml This contains all the pins.

**Comment by zerothi** [comment]
> You can add your requirement constraints to the `meta.yaml` but should also add plain `- numpy` entries for things that have a global pinning. Then `conda smithy` will auto-generate a local pinning file for the Python+OS combination in the feedstock in the `.ci_support` directory. Especially for `numpy`, the pinning is a bit tricky to get it working on all OSes and for a wide range of users.
> 
> For the full global pinning, have a look at https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml This contains all the pins.

Thanks. And this is where it gets a bit tricky for me ;)

So you suggest:
```yaml
- host:
  - numpy >=1.13
  - numpy
- run:
  - {{ pin_compatible('numpy', min_pin='1.13') }}
```
or do I not need the double host section?

**Comment by xhochy** [comment]
Yes, one with the pin to match your package and one unpinned for `conda smithy` so that it generates the right pin for the `host` environment. We omit the `numpy >=1.13` in most cases as the conda-forge supported minimal NumPy is typically higher than the one supported by the package.

**Comment by zerothi** [comment]
> Yes, one with the pin to match your package and one unpinned for `conda smithy` so that it generates the right pin for the `host` environment. We omit the `numpy >=1.13` in most cases as the conda-forge supported minimal NumPy is typically higher than the one supported by the package.

Great. Awesome! I'll add this to this pr then so that I don't forget!!! ;)

File: recipe/meta.yaml
Comment (xhochy) [inline recipe/meta.yaml]: This should not include an explicit version number. The build won't be compatible with `1.13`, it will only be compatible with the `numpy` version in `host` and newer ones, ie. `>=1.17` / `>=1.19`  depending on the Python version.
```suggestion
    - {{ pin_compatible('numpy') }}
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Review by xhochy — APPROVED** [review]

**Comment by zerothi** [comment]
@xhochy many thanks for bearing with me! And for clearing out lots of confusing on the conda-forge interplay for me.!

:rocket:

**Pull request merged**
